apps/server/src/modules/h5p-editor/contentStorage/contentStorage.ts

```diff
@@ -1,173 +1,156 @@
+			const fileDeletePromises = fileList.map((file) => this.deleteFile(contentId, file));
 
-			await fsPromises.rmdir(path.join(this.getContentPath(), contentId.toString()));
+			await Promise.all([this.repo.delete(h5pContent), ...fileDeletePromises]);
 		} catch (error) {
-			if (error) {
-				// eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
-				throw new Error(error.toString());
-			}
+			throw new InternalServerErrorException(error, 'ContentStorage:deleteContent');
 		}
 	}
 
 	public async deleteFile(contentId: string, filename: string, user?: IUser | undefined): Promise<void> {
 		this.checkFilename(filename);
-		const absolutePath = path.join(this.getContentPath(), contentId.toString(), filename);
-		const exist = fs.existsSync(absolutePath);
-		if (!exist) {
-			throw new Error('404: Content not Found at deleteFile.');
-		}
-		await fsPromises.rm(absolutePath);
+		const filePath = this.getFilePath(contentId, filename);
+		await this.storageClient.delete([filePath]);
 	}
 
 	public fileExists(contentId: string, filename: string): Promise<boolean> {
 		this.checkFilename(filename);
-		if (contentId === '' || contentId === undefined) {
-			throw new Error('ContentId is empty or undefined.');
-		}
-		const exist = fs.existsSync(path.join(this.getContentPath(), contentId.toString(), filename));
-		const existPromise: Promise<boolean> = <Promise<boolean>>(<unknown>exist);
-		return existPromise;
+
+		const filePath = this.getFilePath(contentId, filename);
+
+		return this.exists(filePath);
 	}
 
 	public async getFileStats(contentId: string, file: string, user: IUser): Promise<IFileStats> {
-		if (!(await this.fileExists(contentId, file))) {
-			throw new Error('404: Content does not found to get file stats.');
+		const filePath = this.getFilePath(contentId, file);
+		const { ContentLength, LastModified } = await this.storageClient.head(filePath);
+
+		if (ContentLength === undefined || LastModified === undefined) {
+			throw new InternalServerErrorException(
+				{ ContentLength, LastModified },
+				'ContentStorage:getFileStats ContentLength or LastModified are undefined'
+			);
 		}
-		const fileStats = await fsPromises.stat(path.join(this.getContentPath(), contentId.toString(), file));
+
+		const fileStats: IFileStats = {
+			birthtime: LastModified,
+			size: ContentLength,
+		};
+
 		return fileStats;
 	}
 
-	public getFileStream(
+	public async getFileStream(
 		contentId: string,
 		file: string,
 		user: IUser,
-		rangeStart?: number | undefined,
-		rangeEnd?: number | undefined
+		rangeStart?: number,
+		rangeEnd?: number
 	): Promise<Readable> {
-		const exist = <boolean>(<unknown>this.fileExists(contentId, file));
-		if (!exist) {
-			throw new Error('404: Content file missing.');
+		const filePath = this.getFilePath(contentId, file);
+
+		if (rangeStart !== undefined && rangeEnd !== undefined) {
```

Comment:
What's with the case that only rangeStart is given? For the temporary files you handled it. Why not here?